### PR TITLE
feat:clean-iptables consider InboundInterceptionMode

### PR DESCRIPTION
**Pull request by cobb-tx**
Signed-off-by: cobb-tx <464693850@qq.com>

**Please provide a description of this PR:**
clean iptables need consider inbound Interception mode

**To help us figure out who should review this PR, please put an X in all the areas that this PR affects.**

- [ ] Ambient
- [ ] Configuration Infrastructure
- [ ] Docs
- [ ] Installation
- [ x ] Networking
- [ ] Performance and Scalability
- [ ] Policies and Telemetry
- [ ] Security
- [ ] Test and Release
- [ ] User Experience
- [ ] Developer Infrastructure

### CODE REVIEW

File: tools/istio-clean-iptables/pkg/cmd/cleanup.go [hunk 1]
```diff
@@ -69,7 +69,12 @@ func removeOldChains(cfg *config.Config, ext dep.Dependencies, cmd string) {
...
 	flushAndDeleteChains(ext, cmd, constants.NAT, chains)
 	// Flush and delete the istio chains from MANGLE table.
-	chains = []string{constants.ISTIOINBOUND, constants.ISTIODIVERT, constants.ISTIOTPROXY}
+	if cfg.InboundInterceptionMode == constants.TPROXY {
+		chains = []string{constants.ISTIOINBOUND, constants.ISTIODIVERT, constants.ISTIOTPROXY, constants.OUTPUT, constants.PREROUTING}
+	} else {
+		chains = []string{constants.ISTIOINBOUND, constants.ISTIODIVERT, constants.ISTIOTPROXY}
+	}
+
 	flushAndDeleteChains(ext, cmd, constants.MANGLE, chains)
```
Comment (howardjohn) [inline tools/istio-clean-iptables/pkg/cmd/cleanup.go:78]: is this going to flush the entire OUTPUT chain?

Comment (cobb-tx) [reply tools/istio-clean-iptables/pkg/cmd/cleanup.go:78]: yes. when  InboundInterceptionMode is TPROXY will build iptables rule
···
* mangle
-A PREROUTING -p tcp -m mark --mark 1337 -j CONNMARK --save-mark
-A OUTPUT -p tcp -m connmark --mark 1337 -j CONNMARK --restore-mark
···

Comment (howardjohn) [reply tools/istio-clean-iptables/pkg/cmd/cleanup.go:78]: But users may have non-istio rules in OUTPUT

Comment (cobb-tx) [reply tools/istio-clean-iptables/pkg/cmd/cleanup.go:78]: yes you are right , i will update pull request

Comment (cobb-tx) [reply tools/istio-clean-iptables/pkg/cmd/cleanup.go]: @howardjohn   i updated , please code review。

**Review by howardjohn — COMMENTED** [review]
Is there any test that can cover this?

**Comment by cobb-tx** [comment]
> Is there any test that can cover this?

@howardjohn  I added the unit tests

**Comment by cobb-tx** [comment]
> Is there any test that can cover this?

yes please code review again。

**Comment by cobb-tx** [comment]
> /retest

File: releasenotes/notes/41431.yaml [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    clean iptables need consider inbound Interception mode.
+    
+    `istio-iptables` build iptables rule when InboundInterceptionMode is TPROXY 
+    like this :
+    `
+    * mangle
+    -A PREROUTING -p tcp -m mark --mark 1337 -j CONNMARK --save-mark
+    -A OUTPUT -p tcp -m connmark --mark 1337 -j CONNMARK --restore-mark
+    `    
+    so clean-iptables need clean this iptables rule
```
Comment (howardjohn) [inline releasenotes/notes/41431.yaml:20]:
```suggestion
    **Fixed** `istio-clean-iptables` to properly cleanup when InboundInterceptionMode is TPROXY.
```

**Review by howardjohn — APPROVED** [review]

**Pull request merged**